homeassistant/components/balboa/__init__.py

```diff
@@ -0,0 +1,121 @@
+"""The Balboa Spa Client integration."""
+import asyncio
+import time
+
+from pybalboa import BalboaSpaWifi
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+import homeassistant.util.dt as dt_util
+
+from .const import (
+    _LOGGER,
+    CONF_SYNC_TIME,
+    DEFAULT_SYNC_TIME,
+    DOMAIN,
+    PLATFORMS,
+    SPA,
+    UNSUB,
+)
+
+BALBOA_CONFIG_SCHEMA = vol.Schema(
+    {vol.Required(CONF_HOST): cv.string, vol.Required(CONF_NAME): cv.string}
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: vol.All(cv.ensure_list, [BALBOA_CONFIG_SCHEMA])}, extra=vol.ALLOW_EXTRA
+)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Configure the Balboa Spa Client component using flow only."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Balboa Spa from a config entry."""
+    host = entry.data[CONF_HOST]
+
+    unsub = entry.add_update_listener(update_listener)
+
+    _LOGGER.debug("Attempting to connect to %s", host)
+    spa = BalboaSpaWifi(host)
+    hass.data[DOMAIN][entry.entry_id] = {SPA: spa, UNSUB: unsub}
+
+    connected = await spa.connect()
+    if not connected:
+        _LOGGER.error("Failed to connect to spa at %s", host)
+        raise ConfigEntryNotReady
+
+    # send config requests, and then listen until we are configured.
+    await spa.send_mod_ident_req()
+    await spa.send_panel_req(0, 1)
+
+    _LOGGER.debug("Starting listener and monitor tasks")
+    hass.loop.create_task(spa.listen())
+    await spa.spa_configured()
+    hass.loop.create_task(spa.check_connection_status())
+
+    # At this point we have a configured spa.
+    forward_setup = hass.config_entries.async_forward_entry_setup
+    for component in PLATFORMS:
+        hass.async_create_task(forward_setup(entry, component))
```

Comment:
```suggestion
    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
```

Reply:
While, yes, that's true, the very next PR will add another platform..

Reply:
This applies regardless of whether it's one platform or one hundred platforms.
`async_forward_entry_setup` take one platform, but `async_setup_platforms` takes an enumerable.

Reply:
ooooh.. ok.. I didn't know that one was available.  Yeah, that makes sense, I'll switch over.  Thanks!